### ISSUE TITLE
Support for rendering methods overrides

### ISSUE THREAD

**Issue by pepelsbey** [docpad/docpad-plugin-marked#11]
Since it’s already [supported by Marked](https://github.com/chjj/marked#overriding-renderer-methods), it would be great to be able to override renderers right from Docpad’s config. It’s also [partially supported by docpad-plugin-robotskirt](https://github.com/docpad/docpad-plugin-robotskirt/#inline). Something like this:

```
plugins: {
marked: {
  markedRenderer: {
    heading: function (text, level) {
      var escapedText = text.toLowerCase().replace(/[^\w]+/g, '-');
      return '<h' + level + '>' +
        text + ' <a name="' +
        escapedText +
        '" class="anchor" href="#' +
        escapedText +
        '"><span class="header-link"># </span></a></h' + level + '>';
    }
  }
},
```

Copied from [StackOverflow question](http://stackoverflow.com/questions/24101070/how-to-override-rendering-methods-of-marked-compiler-under-docpad-plugin-marked) full of additional details.

There’s even a [fork with all needed features applied](https://github.com/vladikoff/docpad-plugin-marked/commit/39cc54a0856c5ee0bc784920a6156acf1e1e5568#diff-c25561cb097f02c31ba4fa2cf5dce74eR34), so it’s just a matter of checking if everything is correct and filing a pull request.

/cc @vladikoff @varya

**Comment by npatmaja** [comment]
@pepelsbey I've created [a fork](https://github.com/npatmaja/docpad-plugin-marked/tree/support-marked-renderer) based on the links you mentioned above. You can try it by pointing your `docpad-plugin-marked` to the branch in your `package.json`:

```
"dependencies": {
    "docpad-plugin-marked": "git+ssh://git@github.com:npatmaja/docpad-plugin-marked.git#support-marked-renderer"
  }
``` 

And to define a custom renderer, put the definition in the `docpadConfig` inside the `docpad.coffee` file.

```
docpadConfig:
  plugins:
    marked:
      markedRenderer:
        heading: (text, level) ->
          "<h#{level}>#{text}</h#{level}>"
``` 
Please let me know your opinion.

**Comment by varya** [comment]
I've created a PR into marked https://github.com/docpad/docpad-plugin-marked/pull/12
Looks like it responds your needs. If so, vote for it :-)

**Comment by kizu** [comment]
#12 was merged, and released as v2.3.0. Thanks, everyone :)
